ledger-api/grpc-definitions/com/daml/ledger/api/v1/transfer_command.proto

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v1;
+
+import "google/protobuf/any.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+
+
+option java_outer_classname = "TransferCommandOuterClass";
+option java_package = "com.daml.ledger.api.v1";
+option csharp_namespace = "Com.Daml.Ledger.Api.V1";
+
+// This message is a WIP and originates from commands.proto
+// TODO reindex
+message TransferCommand {
+  // TODO deprecated, consider dropping it for 3.0
+  // Must correspond to the ledger ID reported by the Ledger Identification Service.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  string ledger_id = ;
+
+  // Identifier of the on-ledger workflow that this command is a part of.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  string workflow_id = ;
+
+  // Uniquely identifies the application or participant user that issued the command.
+  // Must be a valid ApplicationIdString (as described in ``value.proto``).
+  // Required unless authentication is used with a user token or a custom token specifying an application-id.
+  // In that case, the token's user-id, respectively application-id, will be used for the request's application_id.
+  string application_id = ;
+
+  // Uniquely identifies the command.
+  // The triple (application_id, party + act_as, command_id) constitutes the change ID for the intended ledger change,
+  // where party + act_as is interpreted as a set of party names.
+  // The change ID can be used for matching the intended ledger changes with all their completions.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Required
+  string command_id = ;
+
+  // Party on whose behalf the command should be executed.
+  // If ledger API authorization is enabled, then the authorization metadata must authorize the sender of the request
+  // to act on behalf of the given party.
+  // Must be a valid PartyIdString (as described in ``value.proto``).
+  // Required
+  string submitter = ;
+
+  reserved 6; // was ledger_effective_time
```

Comment:
@rgugliel-da Do we want to time-bound transfer-out/in commands to the sequencing time similar to how we time-bound the ledger time against the record time for normal transactions? That's our usual strategy for dealing with replay attacks: bind each message to a time and then deduplicate messages during a bounded period.

* If not, we need a different story around transfer replay prevention.
* If so, we probably want to give the user the option to specify the reference time (ledger time) for transfers too.

Reply:
I think we should try to handle both types (usual/current commands and future transfers) as similarly as possible, so I'd be in favor to introduce a ledger_time as well here. WDYT?